qiita_db/support_files/patches/python_patches/25.py

```diff
@@ -0,0 +1,121 @@
+# May 19, 2015
+# We attach the prep template directly to the study. The raw data is no longer
+# attached to the study directly, the prep template points to them. This will
+# make the RawData to be effectively just a container for the raw files,
+# which is how it was acting previously.
+
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.data import RawData
+from qiita_db.util import move_filepaths_to_upload_folder
+
+conn_handler = SQLConnectionHandler()
+queue = "PATCH_25"
+conn_handler.create_queue(queue)
+
+# the system may contain raw data with no prep template associated to it.
+# Retrieve all those raw data ids
+sql = """SELECT raw_data_id
+         FROM qiita.raw_data
+         WHERE raw_data_id NOT IN (
+            SELECT DISTINCT raw_data_id FROM qiita.prep_template);"""
+rd_ids = [x[0] for x in conn_handler.execute_fetchall(sql)]
+
+# We will delete those RawData. However, if they have files attached, we should
+# move them to the uploads folder of the study
+sql_detach = """DELETE FROM qiita.study_raw_data
+                WHERE raw_data_id = %s AND study_id = %s"""
+sql_unlink = "DELETE FROM qiita.raw_filepath WHERE raw_data_id = %s"
+sql_delete = "DELETE FROM qiita.raw_data WHERE raw_data_id = %s"
+move_files = []
+for rd_id in rd_ids:
+    rd = RawData(rd_id)
+    filepaths = rd.get_filepaths()
+    studies = sorted(rd.studies)
+    if filepaths:
+        # we need to move the files to a study. We chose the one with lower
+        # study id. Currently there is no case in the live database in which a
+        # RawData with no prep templates is attached to more than one study,
+        # but I think it is better to normalize this just in case
+        move_files.append((min(studies), filepaths))
```

Comment:
Just to confirm, there's no possible way we would get to this line if there are no studies right? i.e. is there any way that studies could be an empty list? If so, then `min` would error.